packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -812,7 +822,9 @@ function deserializeResponseValue(
   switch (type.type) {
     case "datetime":
       return required
-        ? `new Date(${restValue})`
+        ? type.nullable
+          ? `${restValue} === null ? null : new Date(${restValue})`
+          : `new Date(${restValue})`
```

Comment:
Maybe not related to your change, I just wonder if we should generate some utils to handle nullable and required in general?

Reply:
Yeah, I would like to refactor a bit our serializer/deserializer to be a bit more verbose but easier to read and maintain

Reply:
I think I'll take on that once we don't have a lot of concurrent work happening to minimize conflicts